integration_tests/src/new_account_tests.rs

```diff
@@ -0,0 +1,159 @@
+    // use rsa::{BigUint, RSAPrivateKey};
+    use uuid::Uuid;
+
+    #[test]
+    fn new_account() {
+        let account = generate_account();
+        let folder_id = Uuid::new_v4();
+        let folder_key = AesImpl::generate_key();
+
+        assert_matches!(
+            ClientImpl::new_account(
+                &api_loc(),
+                &account.username,
+                &sign(&account),
+                account.keys.to_public_key(),
+                folder_id,
+                FolderAccessInfo {
+                    folder_id: folder_id,
+                    access_key: aes_key(&folder_key, &folder_key),
+                },
+                rsa_key(&account.keys.to_public_key(), &folder_key)
+            ),
+            Ok(_)
+        );
+    }
+
+    #[test]
+    fn new_account_duplicate() {
+        let account = generate_account();
+        let folder_id = Uuid::new_v4();
+        let folder_key = AesImpl::generate_key();
+
+        assert_matches!(
+            ClientImpl::new_account(
+                &api_loc(),
+                &account.username,
+                &sign(&account),
+                account.keys.to_public_key(),
+                folder_id,
+                FolderAccessInfo {
+                    folder_id: folder_id,
+                    access_key: aes_key(&folder_key, &folder_key),
+                },
+                rsa_key(&account.keys.to_public_key(), &folder_key)
+            ),
+            Ok(_)
+        );
+
+        assert_matches!(
+            ClientImpl::new_account(
+                &api_loc(),
+                &account.username,
+                &sign(&account),
+                account.keys.to_public_key(),
+                folder_id,
+                FolderAccessInfo {
+                    folder_id: folder_id,
+                    access_key: aes_key(&folder_key, &folder_key),
+                },
+                rsa_key(&account.keys.to_public_key(), &folder_key)
+            ),
+            Err(Error::<NewAccountError>::Api(
+                NewAccountError::UsernameTaken
+            ))
+        );
+    }
+
+    #[test]
+    fn new_account_invalid_username() {
+        let account = generate_account();
+        let folder_id = Uuid::new_v4();
+        let folder_key = AesImpl::generate_key();
+
+        assert_matches!(
+            ClientImpl::new_account(
+                &api_loc(),
+                &(account.username.clone() + " "),
+                &sign(&account),
+                account.keys.to_public_key(),
+                folder_id,
+                FolderAccessInfo {
+                    folder_id: folder_id,
+                    access_key: aes_key(&folder_key, &folder_key),
+                },
+                rsa_key(&account.keys.to_public_key(), &folder_key)
+            ),
+            Err(Error::<NewAccountError>::Api(
+                NewAccountError::InvalidUsername
+            ))
+        );
+    }
+
+    // #[test]
+    // fn new_account_invalid_public_key() {
```

Comment:
?

Reply:
these don't pass yet bc we don't check signatures